types/types.go

```diff
@@ -10,7 +10,7 @@ import (
 type HexNumber big.Int
 
 func (i HexNumber) MarshalJSON() ([]byte, error) {
-	hexNumber := fmt.Sprintf("\"0x%x\"", (*big.Int)(&i).Uint64())
```

Comment:
So if I understand correctly this will give wrong number for anything greater then 2^64 -1 ?

Reply:
I think you are right, because it mostly works well, in case of `meter` chain I saw that every HexNumber field was parsed correctly except for a value, and the value was pretty huge